src/main/java/org/folio/list/services/refresh/ListRefreshService.java

```diff
@@ -115,12 +121,36 @@ private int importQueryResults(ListEntity list, UUID queryId) {
       list.getInProgressRefreshId().map(UUID::toString).orElse("NONE"));
     DataBatchCallback dataBatchCallback = dataBatchCallbackSupplier.get();
     int offset = 0;
-    List<List<String>> ids = queryClient.getSortedIds(queryId, offset, DEFAULT_BATCH_SIZE);
-    while (!CollectionUtils.isEmpty(ids)) {
-      offset += ids.size();
-      dataBatchCallback.accept(list, ids);
-      ids = queryClient.getSortedIds(queryId, offset, DEFAULT_BATCH_SIZE);
-    }
+    UUID entityTypeId = list.getEntityTypeId();
+    EntityType entityType = entityTypeClient.getEntityType(entityTypeId);
+    List<String> resultIdName = getIdColumnNames(entityType);
+    List<Map<String, Object>> queryContent;
+    do {
+      QueryDetails queryClientQuery = queryClient.getQuery(queryId, true, offset, DEFAULT_BATCH_SIZE);
+      queryContent = queryClientQuery.getContent();
+      List<List<String>> resultIds = queryContent.stream()
+        .map(row -> {
+          List<String> resultIdValues = new ArrayList<>();
+          for (String resultIdComponent : resultIdName) {
+            resultIdValues.add(row.get(resultIdComponent) != null ? row.get(resultIdComponent).toString() : null);
```

Comment:
I'd probably simplify this with something like this:
```suggestion
            resultIdValues.add(Objects.toString(row.get(resultIdComponent)));
```
That will do "null" instead of `null`, but that seems okay (if it's not, then maybe `Objects.toString(row.get(resultIdComponent), null)`)

There's just a lot of boilerplate there + 2 identical calls (`.get()`), so we might as well clean it up a bit

Reply:
@mweaver-ebsco  done